torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -1222,43 +1222,54 @@ Tensor masked_scatter_backward(const Tensor & grad, const Tensor & mask, IntArra
   return mask_selected.view(sizes);
 }
 
-Tensor cholesky_jvp(const Tensor& input_tangent, const Tensor& L, bool upper) {
+Tensor cholesky_jvp(const Tensor& dA, const Tensor& L, bool upper) {
   at::NoTF32Guard disable_tf32;
-  // Differentiation of the Cholesky decomposition, Iain Murray
-  // https://arxiv.org/abs/1602.07527
-  // equation 8
-  auto input_tangent_ = upper ? input_tangent.mH() : input_tangent;
+  // Let A = LL^H
+  // dA = dLL^H + L(dL)^H
+  // L^{-1}dA(L^{-H}) = L^{-1}dL + (L^{-1}dL)^H
+  //               = sym(L^{-1}dL)
+  // where sym(X) = X + X^H
+  // A short computaiton gives that the inverse of sym is given by
+  // \pi(X) = X.tril() - 0.5*diag(X)
+  // so
+  // dL = L\pi(L^{-1}dA(L^{-H}))
+  TORCH_CHECK(at::allclose(dA, dA.mH(), /*rtol=*/1e-2, /*atol=*/1e-2),
+              "cholesky_jvp: The tangent part of the matrix A should also be ",
+              (dA.is_complex() ? "Hermitian" : "symmetric."));
```

Comment:
This check is not enforced in the primal calculation and it doesn't feel right (because of performance) to include it in the forward AD rule.

Reply:
It can potentially do so if they pass a triangular matrix (cc @IvanYashchuk) but perhaps we could assume it is symmetric and roll with that?
I put it for consistency with `eigh_jvp`, but I do agree it's not the best. Should I remove it from here and also put up a PR that removes it from eigh_jvp?

Reply:
I would probably kill it, but let's ask @soulitzer whether it is needed and whether forward AD can be potentially broken by the user.

Reply:
fwiw, I currently vote for removing these checks as well. My reasoning is that the forward function needs a symmetric (and positive definite) matrix, so the user needs to map the input into a SPD matrix, in particular making the forward AD tensor `dA` symmetric. If they don't do that, even if the matrix is symmetric, it would be a symmetric matrix living in R^{n x n}, and that's not OK (you cannot expect correct forward AD / gradients in that scenario).

Reply:
Right, maybe it is worth mentioning somewhere that `A + dA` is supposed to be of the same structure as `A`...

Reply:
You probably mean that `dA` is symmetric / Hermitian (`A + dA` needn't be SPD). I've just left a comment noting this.

Reply:
I think the original idea behind these checks was that: "If autograd computes something, it has to be correct*".
And so if the forward/backward AD formula only works for a subset of inputs, then it should make sure to raise an error for any input that is valid for f but for which we cannot compute gradients.



* According to our own rules, notably, if the input is not valid for f, then we can do whatever.

Reply:
Offline discussion with @albanD:
The rules for autograd are:
>To try and reduce the impact of this limitation, we define the gradients of the elementary operations by applying the following rules in order:
>* If the function is not defined (:math:`\sqrt(-1)` or :math:`\log(-1)` for example) then this is undefined. Most function will return :math:`nan`, but for performance reasons, some functions will return not-nan values (:math:`\log(-1)` for example).
>* If a gradient exist at the current point, return it.
>* If the function is convex, return one of it's subgradients at the current point, if more than one exists, return the subgradient with minimum norm (as it is always a descent direction, see "Convex Optimization Algorithms" by Dimitri Bertsekas, Proposition 4.3.1 for a proof).
>* Define the gradient at the current point by continuity (note that :math:`inf` is possible here). If multiple values are possible, pick one arbitrarily.

This case falls under the first one (the function is not defined if the input matrix `dA` is not symmetric, as you're calling the `jvp` function wrong) so we can return whatever.